app/services/LocationService.js

```diff
@@ -380,73 +363,86 @@ export default class LocationServices {
-        //       [
-        //         {
-        //           text: languages.t('label.yes'),
-        //           onPress: () => {
-        //             if (isPlatformAndroid()) {
-        //               // showLocationSettings() only works for android
-        //               BackgroundGeolocation.showLocationSettings();
-        //             } else {
-        //               Linking.openURL('App-Prefs:Privacy'); // Deeplinking method for iOS
-        //             }
-        //           },
-        //         },
-        //         {
-        //           text: languages.t('label.no'),
-        //           onPress: () => console.log('No Pressed'),
-        //           style: 'cancel',
-        //         },
-        //       ],
-        //     ),
-        //   1000,
-        // );
-      } else if (!status.authorization) {
-        // we need to set delay or otherwise alert may not be shown
-        // setTimeout(
-        //   () =>
-        //     Alert.alert(
-        //       languages.t('label.require_location_information_title'),
-        //       languages.t('label.require_location_information_message'),
-        //       [
-        //         {
-        //           text: languages.t('label.yes'),
-        //           onPress: () => BackgroundGeolocation.showAppSettings(),
-        //         },
-        //         {
-        //           text: languages.t('label.no'),
-        //           onPress: () => console.log('No Pressed'),
-        //           style: 'cancel',
-        //         },
-        //       ],
-        //     ),
-        //   1000,
-        // );
+      if (!status.locationServicesEnabled || !status.authorization) {
+        setTimeout(() => LocationServices.alertRequiredLocationInfo(), 1000);
       }
     });
   }
 
+  static alertRequiredLocationInfo() {
+    Alert.alert(
+      languages.t('label.require_location_services_title'),
+      languages.t('label.require_location_services_message'),
+      [
+        {
+          text: languages.t('common.yes'),
+          onPress: () => {
+            if (isPlatformAndroid()) {
+              // showLocationSettings() only works for android
+              BackgroundGeolocation.showLocationSettings();
+            } else {
+              // eslint-disable-next-line
+              Linking.openURL('App-Prefs:Privacy'); // Deeplinking method for iOS
+            }
+          },
+        },
+        {
+          text: languages.t('common.no'),
+          onPress: () => {
+            console.log('No Pressed'), LocationServices.stop();
+          },
+          style: 'cancel',
+        },
+      ],
+    );
+  }
+
+  static async checkIfAlreadyRunning() {
+    return new Promise((resolve, reject) => {
+      BackgroundGeolocation.checkStatus(({ isRunning }, error) => {
+        if (error) {
+          reject(error);
+        } else {
+          resolve(isRunning);
+        }
+      });
+    });
+  }
+
+  static async isHeadlessTaskRunning() {
```

Comment:
```suggestion
  static isHeadlessTaskRunning() {
```

Since we aren't using the `await` keyword this method and `checkIfAlreadyRunning` do not need `async`. Check out the Return value of the [MDN async docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function)